src/sbuf.c

```diff
@@ -1235,6 +1236,44 @@ static bool setup_tls(struct tls_config *conf, const char *pfx, int sslmode,
 	return true;
 }
 
+static bool tls_config_changed(struct tls_config *new_server_connect_conf)
+{
+	return (strings_equal(new_server_connect_conf->ca_file, server_connect_conf->ca_file) &&
+		strings_equal(new_server_connect_conf->ca_path, server_connect_conf->ca_path) &&
+		strings_equal(new_server_connect_conf->ca_mem, server_connect_conf->ca_mem) &&
+		(new_server_connect_conf->ca_len == server_connect_conf->ca_len) &&
+		strings_equal(new_server_connect_conf->ciphers, server_connect_conf->ciphers) &&
+		(new_server_connect_conf->ciphers_server == server_connect_conf->ciphers_server) &&
+		(new_server_connect_conf->dheparams == server_connect_conf->dheparams) &&
+		(new_server_connect_conf->ecdhecurve == server_connect_conf->ecdhecurve) &&
+		strings_equal(new_server_connect_conf->ocsp_file, server_connect_conf->ocsp_file) &&
+		strings_equal(new_server_connect_conf->ocsp_mem, server_connect_conf->ocsp_mem) &&
+		(new_server_connect_conf->ocsp_len == server_connect_conf->ocsp_len) &&
+		(new_server_connect_conf->protocols == server_connect_conf->protocols) &&
+		(new_server_connect_conf->verify_cert == server_connect_conf->verify_cert) &&
+		(new_server_connect_conf->verify_client == server_connect_conf->verify_client) &&
+		(new_server_connect_conf->verify_depth == server_connect_conf->verify_depth) &&
+		(new_server_connect_conf->verify_name == server_connect_conf->verify_name) &&
+		(new_server_connect_conf->verify_time == server_connect_conf->verify_time));
+}
+
+static bool skip_tag_pools_dirty(struct tls_config *new_server_connect_conf)
```

Comment:
nit, let's flip this around to avoid a double negation in the actual comparison and call it something `tls_changes_requires_reconnect`

Reply:
I like it thanks for the feedback on the double negation

Reply:
[44b913b](https://github.com/pgbouncer/pgbouncer/pull/1157/commits/44b913b7131abfeb20c05c3e09d0f629d096eb4f)